### PR TITLE
[Popover][Dropdown] Portals

### LINKED ISSUE #395: [Popover] break out of scroll container
As a mineral-ui consumer, I want Dropdowns, Popovers, etc to be able to extend beyond their bounding containers even when `overflow:hidden` type styles are applied, so that I am not constrained in my designs.

AC: 

* `usePortal` prop to render popover content to body

Other considerations:

* Needed to break out of scroll container, e.g. CAAC nav menu
* keyboard nav
* a11y
* works with react 15 and 16

### PR DESCRIPTION
**Pull request by brentertz**
### Description

* Add `usePortal` prop to `Popover` and `Dropdown` - allows rendering of content to body instead of as sibling to trigger
* Extract reusable `Portal` component

### Motivation and context

Using a portal will allow `Popovers`, `Dropdowns`, `Tooltips`, etc to visually "break out" of bounding containers that have `overflow: hidden` styles applied.  It is needed particularly in the case where the component lives inside of a scrollable container.  

closes #395

### Screenshots, videos, or demo, if appropriate

https://395-popover-portal--mineral-ui.netlify.com/

_Unfortunately, the Netlify deploy preview appears to be caching the old version of Popper, so it doesn't look like it is working and there does not appear to be a way to clear the cache manually._

### How to test

1. Visit demo - `Popover` and `Dropdown` components
2. Note the behavior of the `Portal` examples

### Types of changes

- New feature (non-breaking change which adds functionality)

### Checklist

* [x] Renders and functions properly in [all supported browsers](https://github.com/mineral-ui/mineral-ui#browser-support)
* [x] Automated tests written and passing
* [x] [Accessibility](http://webaim.org/intro) and [inclusivity](https://24ways.org/2016/what-the-heck-is-inclusive-design/) considered
* [x] Rendering performance (initial load time & 60fps) and [perceived performance](http://blog.teamtreehouse.com/perceived-performance) considered
* [x] Documentation created or updated
* [x] Tested in [sandbox](https://github.com/facebookincubator/create-react-app#getting-started) if new component or breaking change - **[n/a]**

### CODE REVIEW

**Comment by brentertz** [comment]
Curious as to the team's thoughts on where `Portal` and `EventListener` components live.  Should they be siblings to the other Mineral components and simply not exposed publicly via the index?  This is how things are are currently.  Or should they perhaps live elsewhere - e.g. `/src/utils`, `/src/internal`, `/src/private`, etc.

File: src/Popover/Popover.js [hunk 1]
```diff
@@ -199,21 +190,28 @@ export default class Popover extends Component<Props, State> {
       <Root {...rootProps}>
         <PopoverTrigger {...popoverTriggerProps} />
         {isOpen && popoverContent}
+        {isOpen && (
+          <EventListener
```
Comment (kylegach) [inline src/Popover/Popover.js]: 😍

**Comment by kylegach** [comment]
I think keeping them directly in `/src` is fine for now. If we end up with a bunch more of these internal-only components, we can consider a better location later.

**Comment by brentertz** [comment]
This PR has been stripped down to only include the Portal related updates.  Other updates were extracted and merged separately via https://github.com/mineral-ui/mineral-ui/pull/420

**Pull request merged**